### PR TITLE
solana: additional binaries

### PR DESCRIPTION
**Pull request by riptl**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Now that `rustfmt` is in the tree, additional binaries can be compiled, most notably the Solana validator.

### CODE REVIEW

File: Formula/solana.rb [hunk 1]
```diff
@@ -21,10 +21,12 @@ class Solana < Formula
 
   depends_on "protobuf" => :build
   depends_on "rust" => :build
+  depends_on "rustfmt" => :build
```
Comment (SMillerDev) [inline Formula/solana.rb:24]: Why is this needed? Shouldn't the source be formatted correctly already?

Comment (terorie) [reply Formula/solana.rb:24]: @SMillerDev The code base uses [prost](https://github.com/tokio-rs/prost), a Protobuf generator for Rust. Rust seems to format all generated code before compiling. I think the point is to allow deterministic line numbers in backtraces for auto-generated files.

**Comment by terorie** [comment]
I think the build timed out because it took so long. This is most likely due to compiling every binary separately with `cargo install`. The build script would save a lot of time doing `cargo build --release --all` and then `bin.install`-ing the binaries from `./target/release/`.
So is this a reasonable excuse to not use `cargo install` here?

File: Formula/solana.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/solana.rb:24]: Can you ask upstream for an option to skip this?

**Comment by SMillerDev** [comment]
>  So is this a reasonable excuse to not use cargo install here?

Because all software in homebrew is build from source in our CI.

**Comment by terorie** [comment]
> Because all software in homebrew is build from source in our CI.

@SMillerDev Indeed, but using `cargo build` or `cargo install` would both build from CI. The former would accelerate CI builds by ~8x using parallel builds, but the latter appears to be common practice in Homebrew's Rust formulas.
`cargo install` is slower in this case because it tends to not work well with multi-crate workspaces. Hence I'm asking to change around the build script a little to use `cargo build` + `bin.install` instead of `cargo install`.

**Comment by SMillerDev** [comment]
Reading https://doc.rust-lang.org/cargo/commands/cargo-build.html and https://doc.rust-lang.org/cargo/commands/cargo-install.html, the install seems to be meant for installing packages and the `build` for local development.

**Comment by terorie** [comment]
Closing since I have not been able to come up with a solution. Thank you for the help.
